libs/services/src/view/view.service.ts

```diff
@@ -0,0 +1,99 @@
+import { GovernanceVotingPower } from '@libs/entities/entities/governance.voting.power';
+
+@Injectable()
+export class ViewService {
+  constructor(
+    private readonly vmQueryService: VmQueryService,
+    private readonly cacheService: CacheService,
+    private readonly commonConfigService: CommonConfigService,
+    private readonly governanceContractService: GovernanceContractService,
+  ) { }
+
+  async getGovernanceConfig(): Promise<GovernanceConfig> {
+    return await this.cacheService.getOrSet(
+      CacheInfo.GovernanceConfig.key,
+      async () => await this.getGovernanceConfigRaw(),
+      CacheInfo.GovernanceConfig.ttl,
+    );
+  }
+
+  async getGovernanceConfigRaw(): Promise<GovernanceConfig> {
+    const vmQueryResponse = await this.vmQueryService.query(new VmQueryArguments({
+      contractAddress: this.commonConfigService.config.governance.contractAddress,
+      functionName: 'viewConfig',
+    }));
+
+    return GovernanceConfig.fromVmQueryResponse(vmQueryResponse);
+  }
+
+  async getProposals(from: number, to: number): Promise<any> {
+    const proposals = await this.cacheService.getOrSet(
+      CacheInfo.GovernanceProposals.key,
+      async () => await this.getProposalsRaw(),
+      CacheInfo.GovernanceProposals.ttl,
+    );
+
+    return proposals.slice(from, to);
+  }
+
+  async getProposalsRaw(): Promise<GovernanceProposal[]> {
+    const config = await this.getGovernanceConfig();
+    const proposals: GovernanceProposal[] = [];
+    for (let i = 0; i < (config.lastProposalNonce ?? 0); i++) {
+      proposals.push(await this.getProposalDetails(i));
+    }
+
+    return proposals;
+  }
+
+  async getProposalDetails(proposalNonce: number): Promise<GovernanceProposal> {
+    return await this.cacheService.getOrSet(
+      CacheInfo.GovernanceProposal(proposalNonce).key,
+      async () => await this.getProposalDetailsRaw(proposalNonce),
+      CacheInfo.GovernanceProposal(proposalNonce).ttl,
+    );
+  }
+
+  async getProposalDetailsRaw(proposalNonce: number): Promise<GovernanceProposal> {
+    const vmQueryResponse = await this.governanceContractService.viewProposal(proposalNonce);
+
+    return GovernanceProposal.fromVmQueryResponse(vmQueryResponse);
+  }
+
+  async getAddressVotingPower(address: string): Promise<GovernanceVotingPower> {
+    return await this.cacheService.getOrSet(
+      CacheInfo.GovernanceAddressVotingPower(address).key,
+      async () => await this.getAddressVotingPowerRaw(address),
+      CacheInfo.GovernanceAddressVotingPower(address).ttl,
+    );
+  }
+
+  async getAddressVotingPowerRaw(address: string): Promise<GovernanceVotingPower> {
+    const vmQueryResponse = await this.governanceContractService.viewVotingPower(address);
+
+    return GovernanceVotingPower.fromVmQueryResponse(vmQueryResponse);
+  }
+
+  async getDelegatedVotingInfo(address: string, delegatedAddress: string): Promise<any> {
```

Comment:
use `GovernanceDelegatedVoteInfo` instead of any

Reply:
done